Reflection_Engine/Modify/SetPropertyValue.cs

```diff
@@ -102,6 +102,18 @@ public static object SetPropertyValue(this object obj, string propName, object v
                     }   
                 }
 
+                if (propType == typeof(DateTime) && value is string)
```

Comment:
What if someone tries to set an int as the DateTime using a number of seconds since 1970 rather than a string? Is that a use case we want to consider?

Reply:
I was sure someone would come in and ask for more than this PR wants to cover 😋 

I purposefully kept the scope of this PR as small as possible because this can become a fairly complex problem. I actually did an initial version of this same change a few weeks ago and threw it away because it was getting increasingly complicated. 

So the point of this PR is to do a very small improvement on the existing code without trying to solve the whole thing. As I said in the original comment, the problem doesn't really need to be solved for anything but Excel VBA. So  I don't think there is value in investing too much time in it until proven otherwise.